resources/splunk_app.rb

```diff
@@ -135,11 +130,12 @@ class CustomApp < SplunkApp
   resource_name :splunk_app_custom
 
   action :install do
-    return unless !version || changed?(:version)
+    return unless !new_resource.version || changed?(:version)
 
     directory app_path.to_s do
       user current_owner
       group current_owner
+      recursive true
```

Comment:
Out of curiosity... what issues were you seeing here? 

I think the behavior of `recursive true` can be a bit surprising if not expected in that it at least used to be that created intermediate directories have the default user/group owners... only the end directory has the specified user/group owner.

Reply:
So I'm sure I've encountered some case where the parent directory was missing, but I can't figure out what that case is now since all of the splunk standard directories are provided by the installation. I'm inclined to leave this here because of uncertainty, but also to smooth over the use case of specifying a different base directory for the app.